### PR TITLE
remove default report paths

### LINKED ISSUE #267: permission denied on reports
Hi folks,

Is possible put a ERROR log to inform the user that the plugin failed to read the reports because permission issues? I had to lost several hours until discover that my report analysis have been failing to submit because permission issues, not because it was not found! My files were in a directory created by the root user and sonar is running with a normal user. :'(

I'm running the sonar-runner with -X and I didn't see any log relating this. Is possible include this?

Thanks!

### LINKED ISSUE #331: Get rid of default values for the sonar.cxx.<reportsource>.reportPath
I consider those values to be of no real help, introducing them was probably a dumb idea. Also, once we have no default values anymore, we could print warnings into the log if a value is set and yields no files, which is valuable user feedback.

### PR DESCRIPTION
**Pull request by guwirth**
* remove  DEFAULT_REPORT_PATH from all sensors
* don't execute sensor if key does not exist in configuration (exception is coverage: sensor set coverage to 0 in this case)
* improve error messages and warnings in case of wrong report settings or access problems
* close #267
* close #331

### CODE REVIEW

**Comment by guwirth** [comment]
Ready to review.

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java [hunk 1]
```diff
@@ -72,40 +68,55 @@ public CxxCoverageSensor(Settings settings, FileSystem fs, ProjectReactor reacto
     parsers.add(new VisualStudioParser());
   }
 
+  @Override
+  public boolean shouldExecuteOnProject(Project project) {
+    return fs.hasFiles(fs.predicates().hasLanguage(CxxLanguage.KEY));
```
Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: does this works in multi language?

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java [hunk 2]
```diff
@@ -72,40 +68,55 @@ public CxxCoverageSensor(Settings settings, FileSystem fs, ProjectReactor reacto
...
    * {@inheritDoc}
    */
   @Override
   public void analyse(Project project, SensorContext context) {
 
-    CxxUtils.LOG.debug("Parsing coverage reports");
-    List<File> reports = getReports(conf, reactor.getRoot().getBaseDir().getAbsolutePath(), REPORT_PATH_KEY, DEFAULT_REPORT_PATH);
-    if (reports.isEmpty()) {
-        reports = getReports(conf, fs.baseDir().getPath(), REPORT_PATH_KEY, DEFAULT_REPORT_PATH);
+    Map<String, CoverageMeasuresBuilder> coverageMeasures = null;
```
Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: exactly this is actually going to slow down the local analysis

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java (hunk 1 above)
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: This is a copy of the already existing CxxReportSensor.java / shouldExecuteOnProject: means no change to old version.

**Comment by jmecosta** [comment]
yep this does unnecessary stuff. but once this is merged than i can prepare a pr for the other issue.

looks good to me

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java (hunk 2 above)
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: All other sensor with reports are looking first if there is the report key in the sonar-project.properties if not the sensor is not executed. Coverage is different for two reasons:
1) There are three sonar-project.properties keys and not only one.
2) If no report is available there is still the 'sonar.cxx.coverage.forceZeroCoverage'

Inside of analyse I changed things. Each key is checked and if not available nothing is will be done. If also sonar.cxx.coverage.forceZeroCoverage is false then analyse is doing nothing and should be fast.

Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: It was fast that's true. So I guess the other can also be close as it does not affect performance that much

**Pull request merged**